ostd/src/io_mem.rs

```diff
@@ -35,15 +41,45 @@ impl IoMem {
     /// - Reading from or writing to I/O memory regions may have side effects. Those side effects
     ///   must not cause soundness problems (e.g., they must not corrupt the kernel memory).
     pub(crate) unsafe fn new(range: Range<Paddr>) -> Self {
+        let first_page_start = range.start.align_down(PAGE_SIZE);
+        let last_page_end = range.end.align_up(PAGE_SIZE);
+        let mut new_kva = Kva::<Untracked>::new(last_page_end - first_page_start);
+
+        cfg_if! {
+            if #[cfg(all(feature = "cvm_guest", target_arch = "x86_64"))] {
+                let priv_flags = if tdx_guest::tdx_is_enabled() {
+                    PrivilegedPageFlags::SHARED
+                } else {
+                    PrivilegedPageFlags::empty()
+                };
+            } else {
+                let priv_flags = PrivilegedPageFlags::empty();
+            }
+        }
+
+        let prop = PageProperty {
+            flags: PageFlags::RW,
+            cache: CachePolicy::Uncacheable,
```

Comment:
I think it's up to the driver whether to disable caching. Also the driver may want to control about the permissions. So perhaps `IoMem::new()` should take parameters `PageFlags` and `CachePolicy.`